CMakeLists.txt

```diff
@@ -497,6 +495,29 @@ if (BUILD_TESTING)
     ########################## configure unit tests ############################
     ############################################################################
 
+    # CTest configuration variables need to be set before include(CTest) is called
+    set(VALGRIND_DEFAULT " \
+        --leak-check=full \
+        --leak-resolution=high \
+        --trace-children=yes \
+        -q --error-exitcode=123 \
+        --error-limit=no \
+        --num-callers=40 \
+        --undef-value-errors=no \
+        --log-fd=2 \
+        --suppressions=valgrind.suppressions")
+
+    # "pedantic valgrind" will error on memory that is "Still Reachable". 
+    # We only run this on OpenSSL 1.1.1 because there are hundreds of false positives in other libcryptos.
+    # Tracking issue: https://github.com/aws/s2n-tls/issues/4777
+    if ($ENV{S2N_LIBCRYPTO} MATCHES "openssl-1.1.1")
+        set(MEMORYCHECK_COMMAND_OPTIONS "${VALGRIND_DEFAULT} --run-libc-freeres=yes --errors-for-leak-kinds=all --show-leak-kinds=all")
```

Comment:
Checking S2N_LIBCRYPTO here seems strange to me. This is the same script that finds a libcrypto to link to, but is there any guarantee that S2N_LIBCRYPTO matches that linked libcrypto, or even influenced what libcrypto we tried to link to? Like, this only works because you call configure with a CMAKE_PREFIX_PATH based on S2N_LIBCRYPTO. This feels like a separate script was just copy/pasted into CMarkLists.txt instead of a deliberate cmake design.

But it looks like we currently do that in a couple other places in CMakeLists.txt too, so it's not a new behavior. Maybe just open an issue?

Reply:
I have opened #4814 to track this issue.